cairo-contracts/src/test_contracts/emit_multiple_events_across_contracts.cairo

```diff
@@ -39,3 +39,14 @@ func emit_external{ syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_chec
     IExternalContract.emit_external(contract_address=_external_contract_addr);
     return();
 }
+
+@external
+func emit_sandwich{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
+    let (_external_contract_addr) = external_contract_addr.read();
+    internal.emit();
+    IExternalContract.emit_external(contract_address=_external_contract_addr);
+    internal.emit();
+    IExternalContract.emit_external(contract_address=_external_contract_addr);
```

Comment:
Just to be 100% sure, can you confirm that those two calls will result in two different `CallInfo` inside the `inner_calls` fields? Even if those are calls to the same contract

Reply:
Yep, here is the execution call info:

```
CallInfo {
    call: CallEntryPoint {
        class_hash: Some(
            ClassHash(
                StarkFelt(
                    "0x0279d77db761fba82e0054125a6fdb5f6baa6286fa3fb73450cc44d193c2d37f",
                ),
            ),
        ),
        code_address: None,
        entry_point_type: External,
        entry_point_selector: EntryPointSelector(
            StarkFelt(
                "0x015d40a3d6ca2ac30f4031e42be28da9b056fef9bb7357ac5e85627ee876e5ad",
            ),
        ),
        calldata: Calldata(
            [
                StarkFelt(
                    "0x051a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                ),
                StarkFelt(
                    "0x00d44e726c9388338336a865ac2ad319196e507965cce7fee90b116837b795f7",
                ),
                StarkFelt(
                    "0x0000000000000000000000000000000000000000000000000000000000000000",
                ),
            ],
        ),
        storage_address: ContractAddress(
            PatriciaKey(
                StarkFelt(
                    "0x01a3339ec92ac1061e3e0f8e704106286c642eaf302e94a582e5f95ef5e6b4d0",
                ),
            ),
        ),
        caller_address: ContractAddress(
            PatriciaKey(
                StarkFelt(
                    "0x0000000000000000000000000000000000000000000000000000000000000000",
                ),
            ),
        ),
        call_type: Call,
        initial_gas: 10000000000,
    },
    execution: CallExecution {
        retdata: Retdata(
            [],
        ),
        events: [],
        l2_to_l1_messages: [],
        failed: false,
        gas_consumed: StarkFelt(
            "0x0000000000000000000000000000000000000000000000000000000000000000",
        ),
    },
    vm_resources: ExecutionResources {
        n_steps: 310,
        n_memory_holes: 0,
        builtin_instance_counter: {
            "range_check_builtin": 1,
        },
    },
    inner_calls: [
        CallInfo {
            call: CallEntryPoint {
                class_hash: Some(
                    ClassHash(
                        StarkFelt(
                            "0x0000000000000000000000000000000000000000000000000000000000092000",
                        ),
                    ),
                ),
                code_address: Some(
                    ContractAddress(
                        PatriciaKey(
                            StarkFelt(
                                "0x051a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                            ),
                        ),
                    ),
                ),
                entry_point_type: External,
                entry_point_selector: EntryPointSelector(
                    StarkFelt(
                        "0x00d44e726c9388338336a865ac2ad319196e507965cce7fee90b116837b795f7",
                    ),
                ),
                calldata: Calldata(
                    [],
                ),
                storage_address: ContractAddress(
                    PatriciaKey(
                        StarkFelt(
                            "0x051a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                        ),
                    ),
                ),
                caller_address: ContractAddress(
                    PatriciaKey(
                        StarkFelt(
                            "0x01a3339ec92ac1061e3e0f8e704106286c642eaf302e94a582e5f95ef5e6b4d0",
                        ),
                    ),
                ),
                call_type: Call,
                initial_gas: 10000000000,
            },
            execution: CallExecution {
                retdata: Retdata(
                    [],
                ),
                events: [
                    OrderedEvent {
                        order: 0,
                        event: EventContent {
                            keys: [
                                EventKey(
                                    StarkFelt(
                                        "0x02c99ace14ee0a5ef932dc609df0943ab7ac16b7583634612f8dc35a4289a6ce",
                                    ),
                                ),
                            ],
                            data: EventData(
                                [],
                            ),
                        },
                    },
                    OrderedEvent {
                        order: 2,
                        event: EventContent {
                            keys: [
                                EventKey(
                                    StarkFelt(
                                        "0x02c99ace14ee0a5ef932dc609df0943ab7ac16b7583634612f8dc35a4289a6ce",
                                    ),
                                ),
                            ],
                            data: EventData(
                                [],
                            ),
                        },
                    },
                    OrderedEvent {
                        order: 4,
                        event: EventContent {
                            keys: [
                                EventKey(
                                    StarkFelt(
                                        "0x02c99ace14ee0a5ef932dc609df0943ab7ac16b7583634612f8dc35a4289a6ce",
                                    ),
                                ),
                            ],
                            data: EventData(
                                [],
                            ),
                        },
                    },
                ],
                l2_to_l1_messages: [],
                failed: false,
                gas_consumed: StarkFelt(
                    "0x0000000000000000000000000000000000000000000000000000000000000000",
                ),
            },
            vm_resources: ExecutionResources {
                n_steps: 271,
                n_memory_holes: 0,
                builtin_instance_counter: {},
            },
            inner_calls: [
                CallInfo {
                    call: CallEntryPoint {
                        class_hash: Some(
                            ClassHash(
                                StarkFelt(
                                    "0x0000000000000000000000000000000000000000000000000000000000091000",
                                ),
                            ),
                        ),
                        code_address: Some(
                            ContractAddress(
                                PatriciaKey(
                                    StarkFelt(
                                        "0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                                    ),
                                ),
                            ),
                        ),
                        entry_point_type: External,
                        entry_point_selector: EntryPointSelector(
                            StarkFelt(
                                "0x000a5a5773e9e2a4cae1ba2eb67c56b6f5eaa53170cea3df7e71c057f049e957",
                            ),
                        ),
                        calldata: Calldata(
                            [],
                        ),
                        storage_address: ContractAddress(
                            PatriciaKey(
                                StarkFelt(
                                    "0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                                ),
                            ),
                        ),
                        caller_address: ContractAddress(
                            PatriciaKey(
                                StarkFelt(
                                    "0x051a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                                ),
                            ),
                        ),
                        call_type: Call,
                        initial_gas: 10000000000,
                    },
                    execution: CallExecution {
                        retdata: Retdata(
                            [],
                        ),
                        events: [
                            OrderedEvent {
                                order: 1,
                                event: EventContent {
                                    keys: [
                                        EventKey(
                                            StarkFelt(
                                                "0x03ee3b73373acd583a130924aad6dc38cfdc44ba0555ba94ce2ff63980ea0632",
                                            ),
                                        ),
                                    ],
                                    data: EventData(
                                        [],
                                    ),
                                },
                            },
                        ],
                        l2_to_l1_messages: [],
                        failed: false,
                        gas_consumed: StarkFelt(
                            "0x0000000000000000000000000000000000000000000000000000000000000000",
                        ),
                    },
                    vm_resources: ExecutionResources {
                        n_steps: 46,
                        n_memory_holes: 0,
                        builtin_instance_counter: {},
                    },
                    inner_calls: [],
                    storage_read_values: [],
                    accessed_storage_keys: {},
                },
                CallInfo {
                    call: CallEntryPoint {
                        class_hash: Some(
                            ClassHash(
                                StarkFelt(
                                    "0x0000000000000000000000000000000000000000000000000000000000091000",
                                ),
                            ),
                        ),
                        code_address: Some(
                            ContractAddress(
                                PatriciaKey(
                                    StarkFelt(
                                        "0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                                    ),
                                ),
                            ),
                        ),
                        entry_point_type: External,
                        entry_point_selector: EntryPointSelector(
                            StarkFelt(
                                "0x000a5a5773e9e2a4cae1ba2eb67c56b6f5eaa53170cea3df7e71c057f049e957",
                            ),
                        ),
                        calldata: Calldata(
                            [],
                        ),
                        storage_address: ContractAddress(
                            PatriciaKey(
                                StarkFelt(
                                    "0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                                ),
                            ),
                        ),
                        caller_address: ContractAddress(
                            PatriciaKey(
                                StarkFelt(
                                    "0x051a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                                ),
                            ),
                        ),
                        call_type: Call,
                        initial_gas: 10000000000,
                    },
                    execution: CallExecution {
                        retdata: Retdata(
                            [],
                        ),
                        events: [
                            OrderedEvent {
                                order: 3,
                                event: EventContent {
                                    keys: [
                                        EventKey(
                                            StarkFelt(
                                                "0x03ee3b73373acd583a130924aad6dc38cfdc44ba0555ba94ce2ff63980ea0632",
                                            ),
                                        ),
                                    ],
                                    data: EventData(
                                        [],
                                    ),
                                },
                            },
                        ],
                        l2_to_l1_messages: [],
                        failed: false,
                        gas_consumed: StarkFelt(
                            "0x0000000000000000000000000000000000000000000000000000000000000000",
                        ),
                    },
                    vm_resources: ExecutionResources {
                        n_steps: 46,
                        n_memory_holes: 0,
                        builtin_instance_counter: {},
                    },
                    inner_calls: [],
                    storage_read_values: [],
                    accessed_storage_keys: {},
                },
            ],
            storage_read_values: [
                StarkFelt(
                    "0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02cf",
                ),
            ],
            accessed_storage_keys: {
                StorageKey(
                    PatriciaKey(
                        StarkFelt(
                            "0x0238cf5ef6d6264a50d29a47fdf07ec9b7a8e9873214fa58179c5bb40933fdcb",
                        ),
                    ),
                ),
            },
        },
    ],
    storage_read_values: [],
    accessed_storage_keys: {},
}

```